tests/core/pyspec/eth2spec/test/helpers/random.py

```diff
@@ -90,7 +90,7 @@ def slash_random_validators(spec, state, rng, fraction=0.5):
     for index in range(len(state.validators)):
         # slash at least one validator
         sampled = rng.random() < fraction
-        if index == 0 or sampled:
+        if sampled:
```

Comment:
this fn was written with the intent to have *at least* 1 validator slashed

I'd rather add another parameter, something like `required_slashed_count` and then do the random sampling and then iterate over the set and force them to be slashed so that the total number of vals slashed is `required_slashed_count`

we can just keep `required_slashed_count` as a default param at 1

the new code that requested this change can then explicitly set that param to 0

Reply:
I see. 

> required_slashed_count
but `required_slashed_count` value may be conflicting with `fraction` value.

Alternatively, how about we handle `fraction == 0` case specifically? (The default `fraction` value is `0.5`)
```python
if (index == 0 and fraction != 0) or sampled:
```

Reply:
seems workable but I like having independent knobs we can set without worrying about their interaction